permissions/src/androidMain/kotlin/dev/icerock/moko/permissions/PermissionsControllerImpl.kt

```diff
@@ -5,64 +5,112 @@
+
+    override fun bind(activity: ComponentActivity) {
+        this.activityHolder.value = activity
+        val activityResultRegistryOwner = activity as ActivityResultRegistryOwner
+
+        val key = UUID.randomUUID().toString()
+
+        launcher = activityResultRegistryOwner.activityResultRegistry.register(
+            key,
+            ActivityResultContracts.RequestMultiplePermissions()
+        ) { permissions ->
+            val isCancelled = permissions.isEmpty()
+
+            val permissionCallback = permissionCallback ?: return@register
+
+            if (isCancelled) {
+                permissionCallback.callback.invoke(
+                    Result.failure(RequestCanceledException(permissionCallback.permission))
+                )
+                return@register
+            }
+
+            val success = permissions.values.all { it }
+
+            if (success) {
+                permissionCallback.callback.invoke(Result.success(Unit))
+            } else {
+                if (shouldShowRequestPermissionRationale(permissions.keys.first())) {
+                    permissionCallback.callback.invoke(
+                        Result.failure(DeniedException(permissionCallback.permission))
+                    )
+                } else {
+                    permissionCallback.callback.invoke(
+                        Result.failure(DeniedAlwaysException(permissionCallback.permission))
+                    )
+                }
+            }
+        }
 
         val observer = object : LifecycleEventObserver {
             override fun onStateChanged(source: LifecycleOwner, event: Lifecycle.Event) {
                 if (event == Lifecycle.Event.ON_DESTROY) {
-                    this@PermissionsControllerImpl.fragmentManagerHolder.value = null
+                    this@PermissionsControllerImpl.activityHolder.value = null
                     source.lifecycle.removeObserver(this)
                 }
             }
         }
-        lifecycle.addObserver(observer)
+        activity.lifecycle.addObserver(observer)
     }
 
     override suspend fun providePermission(permission: Permission) {
         mutex.withLock {
-            val fragmentManager: FragmentManager = awaitFragmentManager()
-            val resolverFragment: ResolverFragment = getOrCreateResolverFragment(fragmentManager)
-
             val platformPermission = permission.toPlatformPermission()
             suspendCoroutine { continuation ->
-                resolverFragment.requestPermission(
+                requestPermission(
                     permission,
                     platformPermission
                 ) { continuation.resumeWith(it) }
             }
         }
     }
 
+    private fun requestPermission(
+        permission: Permission,
+        permissions: List<String>,
+        callback: (Result<Unit>) -> Unit
+    ) {
+        permissionCallback = PermissionCallback(permission, callback)
+        launcher?.launch(permissions.toTypedArray())
```

Comment:
we should await launcher instead of just stuck on null value. check old `awaitFragmentManager`